src/Extensions/Startup/OmexApplicationStartup.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Omex.Extensions.Abstractions;
+using Microsoft.Omex.Extensions.Compatability;
+using Microsoft.Omex.Extensions.Logging;
+
+namespace Microsoft.Omex.Extensions.ServiceFabric
+{
+	/// <summary>
+	/// Class to start application with initialized DI and Logging
+	/// </summary>
+	/// <typeparam name="TContext">Service context implentation type used for logging, use NullServiceContext if it's not required</typeparam>
+	public class OmexApplicationStartup<TContext>
+		where TContext : class, IServiceContext
+	{
+		/// <summary>Create an instance of OmexApplicationStartup</summary>
+		/// <param name="serviceTypeName">Name of the service used for logging</param>
+		public OmexApplicationStartup(string? serviceTypeName = null) =>
+			ServiceTypeName = serviceTypeName ?? GetDefaultName();
+
+
+		/// <summary>Run function to execute</summary>
+		/// <param name="function">Function to execute</param>
+		/// <param name="typeRegestration">Action to register types in DI</param>
+		/// <param name="typeResolution">Action to resolve types from DI</param>
+		/// <returns></returns>
+		protected async Task RunAsync(
+			Func<Task> function,
+			Action<IServiceCollection>? typeRegestration = null,
+			Action<IServiceProvider>? typeResolution = null)
+		{
+			try
+			{
+				m_typeRegestration = typeRegestration;
+				m_typeResolution = typeResolution;
+				Task task = function();
+				LogSuccess(Process.GetCurrentProcess().Id, ServiceTypeName);
```

Comment:
>LogSuccess [](start = 4, length = 10)

shouldn't we log success after the task has finished?

Reply:
The task will be running as long as service running, so we should log initialization success, before waiting task